SCons/Tool/vcpkg.py

```diff
@@ -0,0 +1,613 @@
+       the VCPkg builder (though currently, it doesn't actually get built during SCons's normal build phase, since
+       vcpkg currently can't tell us what files will be installed without actually doing the work).
+
+       It includes functionality for enumerating the different kinds of files (headers, libraries, etc.) produced by
+       installing the package."""
+
+    def __init__(self, env, descriptor):
+        super().__init__( descriptor.get_listfile_basename() + ".list", env.Dir('$VCPKGROOT/installed/vcpkg/info/'), env.fs)
+        self.descriptor = descriptor
+        self.loaded = False
+
+    def Headers(self, transitive = False):
+        """Returns the list of C/C++ header files belonging to the package.
+           If `transitive` is True, then files belonging to upstream dependencies of this package are also included."""
+        _vcpkg_print(Debug, str(self.descriptor) + ': headers')
+        files = self.FilesUnderSubPath('include/', transitive)
+        return self.FilesUnderSubPath('include/', transitive)
+
+    def StaticLibraries(self, transitive = False):
+        """Returns the list of static libraries belonging to the package.
+           If `transitive` is True, then files belonging to upstream dependencies of this package are also included."""
+        _vcpkg_print(Debug, str(self.descriptor) + ': static libraries')
+        if self.env.subst('$VCPKGDEBUG') == 'True':
+            return self.FilesUnderSubPath('debug/lib/', transitive)
+        else:
+            return self.FilesUnderSubPath('lib/', transitive)
+
+    def SharedLibraries(self, transitive = False):
+        """Returns the list of shared libraries belonging to the package.
+           If `transitive` is True, then files belonging to upstream dependencies of this package are also included."""
+        _vcpkg_print(Debug, str(self.descriptor) + ': shared libraries')
+        if self.env.subst('$VCPKGDEBUG') == 'True':
+            return self.FilesUnderSubPath('debug/bin/', transitive)
+        else:
+            return self.FilesUnderSubPath('bin/', transitive)
+
+    def FilesUnderSubPath(self, subpath, transitive):
+        """Returns a (possibly empty) list of File nodes belonging to this package that are located under the
+           relative path `subpath` underneath the triplet install directory.
+           If `transitive` is True, then files belonging to upstream dependencies of this package are also included."""
+
+        # Load the listfile contents, if we haven't already. This returns a list of File nodes.
+        if not self.loaded:
+            if not self.exists():
+                raise InternalError(self.get_path() + ' does not exist')
+            _vcpkg_print(Debug, 'Loading ' + str(self.descriptor) + ' listfile: ' + self.get_path())
+            self.files = _read_vcpkg_file_list(self.env, self)
+            self.loaded = True
+
+        triplet = self.descriptor.get_triplet()
```

Comment:
> __[Flake8]__ local variable 'triplet' is assigned to but never used ([view](https://sider.review/gh/repos/104670160/pulls/4226#issue-8146510))
>
> <details>
>
> | Rule |
> | ---- |
> | `F841` |
>
> You can [close this issue](https://sider.review/gh/repos/104670160/pulls/4226#issue-8146510) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).
>
> </details>

`get_triplet` called, then called again. maybe delete this line, or use `triplet` in the next?